x/perpetual/keeper/settle_funding_fee_distribution.go

```diff
@@ -48,48 +31,59 @@ func (k Keeper) SettleFundingFeeDistribution(ctx sdk.Context, mtp *types.MTP, po
 	if mtp.Position == types.Position_LONG {
 		// Ensure liabilitiesLong is not zero to avoid division by zero
 		if liabilitiesLong.IsZero() {
-			return sdk.Coin{}, types.ErrAmountTooLow
+			return types.ErrAmountTooLow
 		}
 		fundingFeeShare = sdk.NewDecFromInt(mtp.Liabilities).Quo(sdk.NewDecFromInt(liabilitiesLong))
 		totalFund = short
 	} else {
 		// Ensure liabilitiesShort is not zero to avoid division by zero
 		if liabilitiesShort.IsZero() {
-			return sdk.Coin{}, types.ErrAmountTooLow
+			return types.ErrAmountTooLow
 		}
 		fundingFeeShare = sdk.NewDecFromInt(mtp.Liabilities).Quo(sdk.NewDecFromInt(liabilitiesShort))
 		totalFund = long
 	}
 
 	// if funding fee share is zero, skip mtp
 	if fundingFeeShare.IsZero() {
-		return sdk.Coin{}, nil
+		return nil
 	}
 
 	// calculate funding fee amount
 	fundingFeeAmount := sdk.NewCoin(baseCurrency, totalFund.Mul(fundingFeeShare).TruncateInt())
-	toPay := sdk.Coin{}
 
-	if balance.Amount.LT(fundingFeeAmount.Amount) {
-		toPay = fundingFeeAmount
-	} else {
-		// transfer funding fee amount to mtp address
-		if err := k.bankKeeper.SendCoins(ctx, fundingFeeCollectionAddress, mtpAddress, sdk.NewCoins(fundingFeeAmount)); err != nil {
-			return sdk.Coin{}, err
-		}
+	// update mtp custody
+	mtp.Custody = mtp.Custody.Add(fundingFeeAmount.Amount)
+
+	// decrease fees collected
+	err = pool.UpdateFeesCollected(ctx, fundingFeeAmount.Denom, fundingFeeAmount.Amount, false)
+	if err != nil {
+		return err
+	}
+
+	// update pool custody balance
+	err = pool.UpdateCustody(ctx, mtp.CustodyAsset, fundingFeeAmount.Amount, true, mtp.Position)
+	if err != nil {
+		return err
+	}
+
+	// update accounted balance for custody side
+	err = pool.UpdateBalance(ctx, mtp.CustodyAsset, fundingFeeAmount.Amount, true, mtp.Position)
```

Comment:
@amityadav0 the pool balance should not be incremented or decremented upon fee distribution, the distribution is also incremented to the custody amount nothing else